File: Ser2023/Company/Company.Web/Modules/ServerTypes/Company/ContactTypeRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface ContactTypeRow {
    ContactTypeId?: number;
    Name?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class ContactTypeRow {
    static readonly idProperty = 'ContactTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Company.ContactType';
    static readonly lookupKey = 'Company.ContactType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ContactTypeRow>('Company.ContactType') }
    static async getLookupAsync() { return getLookupAsync<ContactTypeRow>('Company.ContactType') }

    static readonly deletePermission = 'Company:ContactType:Delete';
    static readonly insertPermission = 'Company:ContactType:Modify';
    static readonly readPermission = 'Company:ContactType:View';
    static readonly updatePermission = 'Company:ContactType:Modify';

    static readonly Fields = fieldsProxy<ContactTypeRow>();
}

File: Ser2023/Company/Company.Web/Modules/Company/ContactType/ContactTypeRow.cs
using Company.Administration.Entities;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace Company.Company;

[ConnectionKey("Company"), Module("Company"), TableName("[person].[ContactType]")]
[DisplayName("Contact Type"), InstanceName("Contact Type")]
[ReadPermission(PermissionKeys.ContactType.View)]
[ModifyPermission(PermissionKeys.ContactType.Modify)]
[DeletePermission(PermissionKeys.ContactType.Delete)]
[LookupScript("Company.ContactType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class ContactTypeRow : LoggingRow<ContactTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Contact Type Id"), Identity, IdProperty]
    public long? ContactTypeId
    {
        get => fields.ContactTypeId[this];
        set => fields.ContactTypeId[this] = value;
    }

    [DisplayName("Name"), Size(50), NotNull, QuickSearch, NameProperty]
    public string Name
    {
        get => fields.Name[this];
        set => fields.Name[this] = value;
    }

    //[DisplayName("Rowguid"), Column("rowguid")]
    //public Guid? Rowguid
    //{
    //    get => fields.Rowguid[this];
    //    set => fields.Rowguid[this] = value;
    //}

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId
    {
        get => fields.TenantId[this];
        set => fields.TenantId[this] = value;
    }

    public Int32Field TenantIdField
    {
        get => fields.TenantId;
    }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive
    {
        get => fields.IsActive[this];
        set => fields.IsActive[this] = value;
    }

    Int16Field IIsActiveRow.IsActiveField
    {
        get => fields.IsActive;
    }

    #endregion Tenant & Activ

    public ContactTypeRow()
        : base()
    {
    }

    public ContactTypeRow(RowFields fields)
        : base(fields)
    {
    }
    public class RowFields : LoggingRowFields
    {
        public Int64Field ContactTypeId;
        public StringField Name;
        //public GuidField Rowguid;

        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}